### PR TITLE
Dehardcode account providers

### PR DESCRIPTION
**Pull request by maximmasterr**
Currently account providers (mojang, elyby and dummy) are hardcoded. This pull request moves each account provider to separate class.

### CODE REVIEW

File: application/MultiMC.cpp [hunk 1]
```diff
@@ -683,6 +684,11 @@ MultiMC::MultiMC(int &argc, char **argv) : QApplication(argc, argv)
         qDebug() << "<> Instances loaded.";
     }
 
+    // laod auth providers
```
Comment (Sebastian-byte) [inline application/MultiMC.cpp:687]: Typo

File: api/logic/minecraft/auth/AuthProviders.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      return m_providers.value(id);
+
+    qDebug() << "Lookup failed";
+    return nullptr;
+  }
+
+  QList<AuthProviderPtr> getAll() {
+    return m_providers.values();
+  }
+}
```
Comment (Sebastian-byte) [inline api/logic/minecraft/auth/AuthProviders.cpp:37]: Add newline at end of file

File: api/logic/minecraft/auth/AuthProviders.h [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+/*!
+ * \brief Namespace for auth providers.
+ * This class main putpose is to handle registration and lookup of auth providers
+ */
+namespace AuthProviders
+{
+  MULTIMC_LOGIC_EXPORT void load();
+  MULTIMC_LOGIC_EXPORT AuthProviderPtr lookup(QString id);
+  MULTIMC_LOGIC_EXPORT QList<AuthProviderPtr> getAll();
+}
```
Comment (Sebastian-byte) [inline api/logic/minecraft/auth/AuthProviders.h:22]: Also here

File: api/logic/minecraft/auth/providers/DummyAuthProvider.h [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+public:
+  QString id()
+  {
+    return "dummy";
+  }
+
+  QString displayName() { return "Local"; }
+  bool dummyAuth() { return true; }
+  QString injectorEndpoint() { return "http://localhost:%1"; };
+};
```
Comment (Sebastian-byte) [inline api/logic/minecraft/auth/providers/DummyAuthProvider.h:25]: Here

File: api/logic/minecraft/auth/providers/ElybyAuthProvider.h [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  QString id()
+  {
+    return "elyby";
+  }
+
+  QString displayName() { return "Ely.by"; };
+  QString injectorEndpoint() { return "ely.by"; };
+  QString authEndpoint() { return "https://authserver.ely.by/auth/"; };
+  QUrl resolveSkinUrl(AccountProfile profile) { return QUrl(((QString) "http://skinsystem.ely.by/skins/%1.png").arg(profile.name)); }
+};
```
Comment (Sebastian-byte) [inline api/logic/minecraft/auth/providers/ElybyAuthProvider.h:26]: Here

File: api/logic/minecraft/auth/providers/MojangAuthProvider.h [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+public:
+  QString id()
+  {
+    return "mojang";
+  }
+
+  QString displayName() { return "Mojang"; };
+  QString authEndpoint() { return "https://authserver.mojang.com/"; };
+  bool canChangeSkin() { return true; };
+};
```
Comment (Sebastian-byte) [inline api/logic/minecraft/auth/providers/MojangAuthProvider.h:25]: And here

**Review by Sebastian-byte — CHANGES_REQUESTED** [review]

**Review by Sebastian-byte — APPROVED** [review]

**Comment by maximmasterr** [comment]
@Sebastian-byte @AfoninZ Please review and do some tests

**Comment by AfoninZ** [comment]
Works for me, I approve

**Review by AfoninZ — APPROVED** [review]

**Pull request merged**